subprojects/maven/src/main/java/org/gradle/api/publish/maven/MavenPomContributor.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.publish.maven;
+
+import org.gradle.api.Incubating;
+import org.gradle.api.provider.Property;
+import org.gradle.api.provider.SetProperty;
+
+import java.util.Map;
+
+/**
+ * A contributor of a Maven publication.
+ *
+ * @since 4.8
+ * @see MavenPom
+ * @see MavenPomContributorSpec
+ */
+@Incubating
+public interface MavenPomContributor {
+
+    /**
+     * The name of this contributor.
+     */
+    Property<String> getName();
+
+    /**
+     * The email
+     */
+    Property<String> getEmail();
+
+    /**
+     * The URL of this contributor.
+     */
+    Property<String> getUrl();
+
+    /**
+     * The organization name of this contributor.
+     */
+    Property<String> getOrganization();
+
+    /**
+     * The organization's URL of this contributor.
+     */
+    Property<String> getOrganizationUrl();
+
+    /**
+     * The roles of this contributor.
+     */
+    SetProperty<String> getRoles();
+
+    /**
+     * The timezone of this contributor.
+     */
+    Property<String> getTimezone();
+
+    /**
+     * The properties of this contributor.
+     */
+    Property<Map<String, String>> getProperties();
```

Comment:
I see a possible breaking change coming here, since we might do something similar to SetProperty (or the other way round, teasing apart SetProperty again). This is probably a fringe use case and so a breaking change in 5.0 probably acceptable

Reply:
I put some thought into that as well. I think we'll likely introduce a special `MapProperty<K, V>` at some point. However, since it will extend `Property<Map<K, V>>` it would at least be source compatible.

Reply:
Fwiw, I do need such a `MapProperty`, and as I'd really like the map to be updateable, I built one based on `SetProperty` (so I don't rely too much on Gradle internals and/or APIs that could break in future versions): https://github.com/tbroyer/gradle-errorprone-plugin/blob/76407271db78c535337c7aa9238fdaf75229b447/src/main/java/net/ltgt/gradle/errorprone/DefaultMapProperty.java
The main reason I'm trying to use a `MapProperty` is because I'd like the property to default to some globally defined options, using the `.set(Provider<…> provider)` of `Property` (rather than convention mapping); but I'd like to keep a DSL that's "updateable" (i.e. where I can add to the map, and not just replace it with another; like one can do with a `ListProperty` or `SetProperty`).

I'd welcome some review of this code, and/or how to build my DSL differently so I wouldn't need that `MapProperty` (the code is currently in a WIP feature branch –that I'll squash and rebase–, so API can change as needed).

Reply:
We plan to add such a `MapProperty` "soon" (see https://github.com/gradle/gradle-native/issues/746 [it's in the native repo because the native team intends to work on it]). Maybe your implementation can be used as starting point?